cli/tests/test_git_fetch.rs

```diff
@@ -1313,6 +1314,7 @@ fn test_fetch_undo_what() {
     ]);
     insta::assert_snapshot!(output, @r"
     ------- stderr -------
+    Warning: `jj op restore --what` is deprecated; use `jj git fetch` afterwards instead
```

Comment:
I think that if you do

```
jj rebase ...
jj git push
jj op restore <before_rebase>
jj git fetch
```

you will end up with a huge mess. Avoiding this mess is the main benefit of `jj op restore --what repo`.